public/app/core/components/Form/Form.tsx

```diff
@@ -0,0 +1,62 @@
+import { css } from '@emotion/css';
+import React, { HTMLProps, useEffect } from 'react';
+import {
+  useForm,
+  Mode,
+  DefaultValues,
+  SubmitHandler,
+  FieldValues,
+  UseFormReturn,
+  FieldErrors,
+  FieldPath,
+} from 'react-hook-form';
+
+export type FormAPI<T extends FieldValues> = Omit<UseFormReturn<T>, 'handleSubmit'> & {
+  errors: FieldErrors<T>;
+};
+
+interface FormProps<T extends FieldValues> extends Omit<HTMLProps<HTMLFormElement>, 'onSubmit' | 'children'> {
+  validateOn?: Mode;
+  validateOnMount?: boolean;
+  validateFieldsOnMount?: FieldPath<T> | Array<FieldPath<T>>;
+  defaultValues?: DefaultValues<T>;
+  onSubmit: SubmitHandler<T>;
+  children: (api: FormAPI<T>) => React.ReactNode;
+  /** Sets max-width for container. Use it instead of setting individual widths on inputs.*/
+  maxWidth?: number | 'none';
+}
+
+export function Form<T extends FieldValues>({
```

Comment:
So is the point of this move that the other one in grafana/ui is deprecated? Do we expect people to have new use cases for the use of this `Form`? Just wondering why the need to duplicate if we will still keep the other deprecated `Form` for the foreseeable future anyway

Reply:
The goal is to remove it from grafana/ui, there's nothing wrong with the component itself, just its dependence on reack-hook-form. That's why I think it's fine to have it in core.

Reply:
Ok that's fair! I guess my main question is what is the plan for deletion? Do we know how many places it is used outside of grafana repo? Just wondering how long we will have this duplicated code.

Reply:
It's not really gonna be duplicated code, this component is already a bit different from the one in grafana/ui and should be used from now on. Deleting the component from grafana/ui will be a bit more challenging as we don't have good data where it can be used, plus deleting it might break some unmaintained plugins which are still used. I'd like to get rid of it with the Grafana 11 release, but that might prove not feasible.

Reply:
Cool! Just wanted to make sure we had some plan to delete that one 👍